### ISSUE TITLE
Default filters no longer working on twitch.tv

### ISSUE THREAD

**Issue by taylorbwilliams2** [uBlockOrigin/uAssets#122]
### Describe the issue

Also posted on the uBO subreddit.
Been using uBO for about 6 months now, no problem on twitch with any of the ads. But just in the past week or two I've seen new ads slipping through. One is a Gildan Underwear commercial with Blake Sheldon, and the other is a Geico commercial featuring players of the TSM league team. A third is an an Old Spice commercial. There could be other ads I'm unaware of, but these are the only ones I've personally seen get through.

### One or more specific URLs where the issue occurs

https://www.twitch.tv/reynad27 (entry ads)
https://www.twitch.tv/nl_kripp (entry ads, occasionally when streamer plays ads between games)
https://www.twitch.tv/trumpsc (similar timing of ads)

### Screenshot in which the issue can be seen

![twitchad](https://cloud.githubusercontent.com/assets/21329305/18077820/db38d772-6e3c-11e6-9ffc-6a2eb2ae471b.png)

![twitchad2](https://cloud.githubusercontent.com/assets/21342704/18098682/1d3b54f0-6e98-11e6-91bf-037e38672362.png)

### Steps for anyone to reproduce the issue

Logging on to any above of twitch URL's while live (sometimes commercial breaks during the past broadcast videos, but not as consistently during live) will play an entry ad of either of the two described above.
I have a few other extensions enabled (Dark Reader, Dark Youtube skin, FrankerZFace, BTTV, RES) but I have still gotten these ads after disabling all but uBO.
I have purged all caches and re-updated all default filters.

### Your settings

- Browser/version: Google Chrome Version 52.0.2743.116 m (current)
- uBlock Origin version: uBlock Origin v1.9.4 (current)

##### Your filter lists

Default filter lists

##### Your custom filters (if any)

None

**Comment by lewisje** [comment]
At least for the moment, the non-default lists and custom filters I mentioned in the ABP forums should work; I tried just a few minutes ago, watching a stream I know to be partnered (that's where you see most of the ads), and the ads didn't show up: https://adblockplus.org/forum/viewtopic.php?p=157432#p157432

This wouldn't apply if there were something new about Twitch, within the past 9 days, to bypass even those filters.

**Comment by taylorbwilliams2** [comment]
It's just applying these to my custom filters, yes?

@@||amazon-adsystem.com^$domain=twitch.tv
@@||fan.twitch.tv/crossdomain.xml
@@||twitch.tv/ads/ads.js|
||chat.betterttv.net/chat/cometchatjs.php?$script,domain=twitch.tv
||d2f8tie0ouoft.cloudfront.net^$object,domain=twitch.tv
||pubsub-edge.twitch.tv^
twitch.tv##.sub-upsell
twitch.tv###sub-upsell

I've applied these, but they didn't seem to help, and I still had the same ads on twitch on the URLs I mentioned in my original post.

**Comment by lewisje** [comment]
From the post I just linked to:

> try subscribing to these lists if you haven't already (You should already be subscribed to EasyList and EasyPrivacy by default):
* [Peter Lowe's Ad Servers](https://pgl.yoyo.org/as/serverlist.php?hostformat=adblockplus&showintro=0&mimetype=plaintext&skip%5B%5D=adf.ly&skip%5B%5D=boldchat.com)
* [Fanboy's Enhanced Tracking List](https://www.fanboy.co.nz/enhancedstats.txt)

That means no, you don't need just those custom filters.

**Comment by taylorbwilliams2** [comment]
Yes sorry, I should have clarified, I was already subscribed to those lists. I'll give it a try and see if I can reproduce any ads with the new filter setup.

**Comment by lewisje** [comment]
> I was already subscribed to those lists

Then why did you say

> Default filter lists

in the first post of this Issue? Those two lists are not checked by default in uBO, even though the checkboxes are there to begin with (as opposed to a custom subscription, which requires its URL to be entered manually by the user to set it up).

---
With that said, what I know is that my own setup does block Twitch ads, and I haven't quite gotten to the bottom of *how*; I'll try later to look into what's doing it.

I don't have any ad-blocking extensions in Chrome other than uBO, and I know from testing uBO for Edge that Twitch ads do show up there, because [extensions for Edge cannot intercept network requests initiated by plugin content](https://github.com/MicrosoftEdge/MicrosoftEdge-Documentation/tree/master/extensions/API-support/supported-APIs#webrequest) (this is also why I've given up on Edge after being so hopeful about its Chrome-inspired extension system); this means that it's really uBO that's doing it in Chrome, and not any other part of my ad-blocking setup (HOSTS file, router setup, etc.).

**Comment by taylorbwilliams2** [comment]
I can see the confusion. I subscribed to them as per instructed from your APB post. One of them was subscribed to in my default uBO settings, but the other was not in the defaults, so I made sure to subscribe. Sorry for the misunderstanding.

My issue now is that I still seem to be getting the ads, on live video and VOD replays. I have a screencap I just took below.

![twitchad3](https://cloud.githubusercontent.com/assets/21342704/18177556/91065662-702f-11e6-9784-bad98173bffe.png)

I've seen a few other people with the very same issue recently with twitch. I'm wondering if it has to do with their gradual shift to HTML 5 players on their sites. Either way, it seems some people's setups (such as yours) are blocking the ads, while others aren't. Seems nobody can peg down exactly why

**Comment by lewisje** [comment]
It could have to do with the HTML5 player; I'm using the Flash player and do not see ads. I'll see whether I can trigger the HTML5 player on my setup.

**Comment by Snapy** [comment]
@lewisje You can do so by using this https://player.twitch.tv/?channel=CHANNELNAME&html5 (replace CHANNELNAME by the relevant info).

**Comment by Darkrai87** [comment]
I am also having this issue, can confirm that I also have the html5 player.

**Comment by taylorbwilliams2** [comment]
Yeah just as an update, I believe I also have thee html5 player, as I opted in during the html5 beta during a month of twitch turbo.

This seems to allow new ads through uBO, ABP, and any other adblockers or filters I've found.

**Comment by gorhill** [comment]
I originally tried all the URLs in the issue report, and they all came as "offline". I tried again, and still all "offline". I am going to need more URLs where ads are occurring -- as long as I can't reproduce, I can't investigate.

**Comment by Hrxn** [comment]
> I originally tried all the URLs in the issue report, and they all came as "offline".
  
Well, this kinda is how Twitch works ;-)

I haven't seen any so far on Twitch, but on the other hand I'm not a heavy user there.
If anyone can post a recent example here I will test it as well..

**Comment by BilBg** [comment]
All the 6 thumbnails/links on this page work now but I don't see any ads:
https://www.twitch.tv/directory/game/Shadowverse

From uBO popup UI (after clicking all the 6 thumbnails, some several times):
requests blocked
on this page
498 or 19%

Other working links (again I don't see any ads):
Right-Click at left in column "Members" on "blue" (on-line?) user and open in new tab:
https://www.twitch.tv/team/tempostorm

Maybe ads show depending on country (per IP of user)?
(this is the first time I try anything on twitch.tv, i.e. I don't know this site)
